File: server/src/main/scala/lassi/server/io/IOService.scala
package lassi.server.io

import lassi.domain.Scene
import lassi.server.settings.Settings

trait IOService {

  def updateSceneFromDisk(scene: Scene): Unit

  def saveSceneToDisk(scene: Scene): Unit

  def downloadScene(scene_id: String): String

  def getAllScenesFromDisk: Seq[Scene]

  def deleteSceneFromDisk(sceneId: String): Unit

  def getSceneFromDisk(sceneId: String): Scene

  def getButtons: Seq[Boolean]

  def readSettings: Settings

  def writeSettings(settings: Settings): Unit
}

File: server/src/main/java/lighting/server/IO/IOService.java
package lighting.server.IO;

import java.io.IOException;
import java.util.List;

import lighting.server.scene.Scene;
import lighting.server.settings.Settings;

public interface IOService {

	void updateSceneFromDisk(Scene scene) throws IOException;

	void saveSceneToDisk(Scene scene) throws IOException;

	String downloadScene(String scene_id) throws IOException;

	List<Scene> getAllScenesFromDisk() throws IOException;

	void deleteSceneFromDisk(String scene_id) throws IOException;

	Scene getSceneFromDisk(String scene_id) throws IOException;

	List<Boolean> getButtons() throws IOException;

	// migrated
	void saveSettingsToDisk(Settings settings) throws IOException;

	// migrated
	Settings getSettingsFromDisk() throws IOException;
}
